protostar/commands/test/environments/factory.py

```diff
@@ -12,23 +12,20 @@
     ExecutionResourcesSummary,
 )
 from protostar.commands.test.starkware.test_execution_state import TestExecutionState
+from protostar.commands.test.test_output_recorder import OutputName
 
 
 async def invoke_setup(
     function_name: str,
     state: TestExecutionState,
-    stdout_buffer: StringIO,
-):
-    env = SetupExecutionEnvironment(state)
-    with redirect_stdout(stdout_buffer):
-        await env.invoke(function_name)
+) -> Optional[ExecutionResourcesSummary]:
+    env = TestExecutionEnvironment(state)
+    return await env.invoke(function_name, "setup")
```

Comment:
Could you please move this "setup" value one more level deeper? If extracting this as a parameter will be needed, one could do it in the future.

I don't want to have any output collection related parameters in the `invoke` method. 😄